rust/src/array.rs

```diff
@@ -0,0 +1,248 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::convert::From;
+use std::iter::Iterator;
+use std::rc::Rc;
+use std::str;
+use std::string::String;
+
+use super::bitmap::Bitmap;
+
+use bytes::{Bytes, BytesMut, BufMut};
+
+pub enum ArrayData {
```

Comment:
Probably you have a plan to make this immutable and add a sort of `ArrayBuilder` in the future. It would be great if it can be commented somewhere.

Reply:
`Bytes` have a nice primitive for that, see [freeze](https://carllerche.github.io/bytes/bytes/struct.BytesMut.html#examples).